cli/admin/src/test/java/io/pravega/cli/admin/AbstractTlsAdminCommandTest.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.cli.admin;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import io.pravega.local.LocalPravegaEmulator;
+import io.pravega.test.common.SecurityConfigDefaults;
+import io.pravega.test.common.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.Timeout;
+
+import java.net.URI;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public abstract class AbstractTlsAdminCommandTest {
+
+    @Rule
+    public final Timeout globalTimeout = new Timeout(80, TimeUnit.SECONDS);
+
+    protected final AtomicReference<AdminCommandState> state = new AtomicReference<>();
+    protected boolean authEnabled = false;
+    protected boolean tlsEnabled = false;
+    LocalPravegaEmulator localPravega;
+
+    // Security related flags and instantiate local pravega server.
+    private final Integer controllerPort = TestUtils.getAvailableListenPort();
+    private final Integer segmentStorePort = TestUtils.getAvailableListenPort();
+    private final Integer restServerPort = TestUtils.getAvailableListenPort();
+
+    @Before
+    public void setUp() throws Exception {
+
+        // Create the secure pravega server to test commands against.
+        LocalPravegaEmulator.LocalPravegaEmulatorBuilder emulatorBuilder = LocalPravegaEmulator.builder()
+                .controllerPort(controllerPort)
+                .segmentStorePort(segmentStorePort)
+                .zkPort(TestUtils.getAvailableListenPort())
+                .restServerPort(TestUtils.getAvailableListenPort())
+                .enableRestServer(true)
+                .restServerPort(restServerPort)
+                .enableAuth(authEnabled)
+                .enableTls(tlsEnabled);
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (authEnabled) {
+            emulatorBuilder.passwdFile("../../config/" + SecurityConfigDefaults.AUTH_HANDLER_INPUT_FILE_NAME)
```

Comment:
The String "../../config" gets repeated many times below. Create a String constant and use it here and below.

Reply:
Stored in a local variable.